pydevd_frame.py

```diff
@@ -504,13 +518,12 @@ def trace_dispatch(self, frame, event, arg):
                             #When we get to the pydevd run function, the debugging has actually finished for the main thread
                             #(note that it can still go on for other threads, but for this one, we just make it finish)
                             #So, just setting it to None should be OK
-                            #I.e.: when setting to note we'll set the state to STATE_RUN and clear any stepping flags.
-                            back_filename, base = GetFilenameAndBase(back)
-                            if base == 'pydevd.py' and back.f_code.co_name == 'run':
+                            base = basename(back.f_code.co_filename)
```

Comment:
You should still keep on using GetFilenameAndBase()... as its contents are usually cached, it should be faster than calling basename on a filename.